### ISSUE TITLE
Adding class condition to time embeddings in resnet block

### ISSUE THREAD

**Issue by ChintanTrivedi** [voletiv/mcvd-pytorch#4]
Referenced paper by [Dhariwal et al. 2021](https://arxiv.org/abs/2105.05233) suggests to use `AdaGN(h, y) = ys GroupNorm(h)+yb` to combine time `ys` and class `yb` embeddings with resnet block activations `h`. I am having some trouble understanding how to implement this in the mcvd code in this repo since class conditioning lines are commented out. It seems time and class embeddings are to be concatenated together (based on commented code) and are fed together to the resnet block as "emb". 
```
# resnetblock 
def forward(self, x, temb=None, yemb=None, cond=None):
    if emb is not None:
        emb = torch.cat([temb, yemb], dim=1) # Combine time and class embeddings
        emb_out = self.Dense_0(self.act_emb(emb))[:, :, None, None]  # Linear projection
        scale, shift = torch.chunk(emb_out, 2, dim=1)
        [ ... ]
        emb_norm = self.Norm_0(x)
        x = emb_norm * (1 + scale) + shift

```
My confusion:- How does splitting the linear projection of the combined embeddings into 2 chunks give us `scale` and `shift`? How to interpret these two values in relation to the time and class embeddings? It seems `scale` might be analogous to `temb` and `shift` to `yemb`, but that's not what the code suggests.

PS: Getting some really good results for prediction tasks, thanks for making your code available!

**Comment by ChintanTrivedi** [comment]
Thanks for the clarifications. I read the literature again and my interpretations of ys and yb were incorrect. For now, I'm using the first embedding option for training.

**Comment by ChintanTrivedi** [comment]
Update: I trained the class-conditioned model using the first option to introduce embeddings, but it doesn't seem to correlate at all. In fact, the generated frames seem to completely ignore the class label. I'll try to add the classifier-gradient method for guided diffusion and see if that helps.
